webpack.base.js

```diff
@@ -55,11 +55,25 @@ module.exports = ({ entry, mode, filename, types, keepBuildEnvVariables }) => ({
     ],
   },
 
-  plugins: !keepBuildEnvVariables
-    ? [
-        new webpack.DefinePlugin({
-          __BUILD_ENV__SDK_VERSION__: JSON.stringify(buildEnv.SDK_VERSION),
-        }),
-      ]
-    : [],
+  plugins: [
+    new webpack.SourceMapDevToolPlugin(
+      mode === 'development'
+        ? // Use an inline source map during development (default options)
+          {}
+        : // When bundling for release, produce a source map file so it can be uploaded to Datadog,
+          // but don't append the source map comment to bundles as we don't upload the source map to
+          // the CDN (yet).
+          {
+            filename: '[file].map',
+            append: false,
+          }
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏what about not generating source map as long as we don't upload it?

Reply:
We do [upload it to Datadog](https://github.com/DataDog/browser-sdk/blob/a41f1df32f9460a638f97f6cc47c1e35b2dd1255/scripts/upload-source-maps.js), so we need to generate the source map. Isn't my comment clear enough?